breaker/breaker.go

```diff
@@ -0,0 +1,29 @@
+package breaker
+
+import (
+	"fmt"
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/rcrowley/go-metrics"
+	"net/http"
+	"time"
+)
+
+func Do(name string, url string) (resp *http.Response, err error) {
+	return DoWithFallback(name, url, nil)
+}
+
+func DoWithFallback(name string, url string, f func(error) error) (resp *http.Response, err error) {
+	err = hystrix.Do(name, func() error {
+		start := time.Now()
+		if resp, err = Get(url); err == nil {
+			measureRequest(start, fmt.Sprintf("planb.breaker.%s", name))
+		}
+		return err
+	}, f)
+	return
+}
+
+func measureRequest(start time.Time, key string) {
+	t := metrics.DefaultRegistry.GetOrRegister(key, metrics.NewTimer).(metrics.Timer)
```

Comment:
my previous comment probably applies to the defaultregistry, too. Maybe, there could be a breaker object, that encapsulates the client, and the metrics, and if those are not set, only then uses the default.